chainercv/links/model/ssd/multibox_loss.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import division
+
+import numpy as np
+
+import chainer
+import chainer.functions as F
+
+
+def _elementwise_softmax_cross_entropy(x, t):
+    assert x.shape[:-1] == t.shape
+    shape = t.shape
+    x = F.reshape(x, (-1, x.shape[-1]))
+    t = F.flatten(t)
+    return F.reshape(
+        F.softmax_cross_entropy(x, t, reduce='no'), shape)
+
+
+def _hard_negative(x, positive, k):
+    xp = chainer.cuda.get_array_module(x, positive)
+    x = chainer.cuda.to_cpu(x)
+    positive = chainer.cuda.to_cpu(positive)
+    rank = (x * (positive - 1)).argsort(axis=1).argsort(axis=1)
+    hard_negative = rank < (positive.sum(axis=1) * k)[:, np.newaxis]
+    return xp.array(hard_negative)
+
+
+def multibox_loss(mb_locs, mb_confs, gt_mb_locs, gt_mb_labels, k):
+    """Computes multibox losses.
+
+    This is a loss function used in [#]_.
+    This function returns :obj:`loc_loss` and :obj:`conf_loss`.
+    :obj:`loc_loss` is a loss for localization and
+    :obj:`conf_loss` is a loss for classification.
+    The formulas of these losses can be found in
+    the equation (2) and (3) in the original paper.
+
+    .. [#] Wei Liu, Dragomir Anguelov, Dumitru Erhan,
+       Christian Szegedy, Scott Reed, Cheng-Yang Fu, Alexander C. Berg.
+       SSD: Single Shot MultiBox Detector. ECCV 2016.
+
+    Args:
+        mb_locs (chainer.Variable): The offsets and scales
```

Comment:
I think it is better to explicitly state that inputs can be either an array or a variable.